x/protorev/keeper/protorev.go

```diff
@@ -135,3 +136,247 @@ func (k Keeper) DeleteAllEntriesForKeyPrefix(ctx sdk.Context, keyPrefix []byte)
+	bz := store.Get(types.KeyPrefixLatestBlockHeight)
+	if bz == nil {
+		// This should never happen as the module is initialized on genesis and reset in the post handler
+		return 0, fmt.Errorf("block height has not been set in state")
+	}
+
+	res := sdk.BigEndianToUint64(bz)
+
+	return res, nil
+}
+
+// SetLatestBlockHeight sets the latest block height that protorev was run on
+func (k Keeper) SetLatestBlockHeight(ctx sdk.Context, blockHeight uint64) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixLatestBlockHeight)
+	store.Set(types.KeyPrefixLatestBlockHeight, sdk.Uint64ToBigEndian(blockHeight))
+}
+
+// ---------------------- Admin Stores  ---------------------- //
+
+// GetAdminAccount returns the admin account for protorev
+func (k Keeper) GetAdminAccount(ctx sdk.Context) (sdk.AccAddress, error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixAdminAccount)
+	bz := store.Get(types.KeyPrefixAdminAccount)
+	if bz == nil {
+		return nil, fmt.Errorf("admin account not found, it has not been initialized through governance")
+	}
+
+	return sdk.AccAddress(bz), nil
+}
+
+// SetAdminAccount sets the admin account for protorev
+func (k Keeper) SetAdminAccount(ctx sdk.Context, adminAccount sdk.AccAddress) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixAdminAccount)
+	store.Set(types.KeyPrefixAdminAccount, adminAccount.Bytes())
+}
+
+// GetDeveloperAccount returns the developer account for protorev
+func (k Keeper) GetDeveloperAccount(ctx sdk.Context) (sdk.AccAddress, error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixDeveloperAccount)
+	bz := store.Get(types.KeyPrefixDeveloperAccount)
+	if bz == nil {
+		return nil, fmt.Errorf("developer account not found, it has not been initialized by the admin account")
+	}
+
+	return sdk.AccAddress(bz), nil
+}
+
+// SetDeveloperAccount sets the developer account for protorev that will receive a portion of arbitrage profits
+func (k Keeper) SetDeveloperAccount(ctx sdk.Context, developerAccount sdk.AccAddress) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixDeveloperAccount)
+	store.Set(types.KeyPrefixDeveloperAccount, developerAccount.Bytes())
+}
+
+// GetMaxRoutesPerTx returns the max number of routes that can be iterated per transaction
+func (k Keeper) GetMaxRoutesPerTx(ctx sdk.Context) (uint64, error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixMaxRoutesPerTx)
+	bz := store.Get(types.KeyPrefixMaxRoutesPerTx)
+	if bz == nil {
+		// This should never happen as it is set to the default value on genesis
+		return 0, fmt.Errorf("max routes configuration has not been set in state")
+	}
+
+	res := sdk.BigEndianToUint64(bz)
+	return res, nil
+}
+
+// SetMaxRoutesPerTx sets the max number of routes that can be iterated per transaction
+func (k Keeper) SetMaxRoutesPerTx(ctx sdk.Context, maxRoutes uint64) error {
```

Comment:
Is this really per Tx? It looks like this limit is applied per swap message, since the ante handler is called once per message.

Reply:
resolved per convo. tldr is the antehandler runs per tx and we aggregate all the pools that were swapped on in the tx across all messages so MaxRoutesPerTx does constrain execution on a per tx scale.